assembl/static2/js/app/components/voteSession/gaugeVoteForProposal.jsx

```diff
@@ -0,0 +1,292 @@
+      const choicesValues = props.choices.reduce((accumulator, item) => {
+        if ('value' in item) {
+          return accumulator.concat(item.value);
+        }
+        return accumulator;
+      }, []);
+      if (choicesValues.length) {
+        this.maximum = Math.max(...choicesValues);
+        this.minimum = Math.min(...choicesValues);
+      }
+    }
+
+    if (props.choices && props.choices.length) {
+      props.choices.forEach((choice) => {
+        this.marks[`${choice.value}`] = {
+          style: markStyle,
+          label: <div>{choice.label}</div>
+        };
+      });
+    }
+  }
+
+  onAfterChange(value: number) {
+    this.setState({
+      value: value
+    });
+    if (this.inputElement && 'value' in this.inputElement) {
+      this.inputElement.value = value;
+    }
+    this.props.voteForProposal(this.props.proposalId, this.props.id, value);
+  }
+
+  render() {
+    return (
+      <div className="gauge-vote-for-proposal">
+        <p>{this.props.instructions}</p>
+        <Slider
+          min={this.minimum}
+          max={this.maximum}
+          marks={this.marks}
+          included={false}
+          step={null}
+          trackStyle={trackStyle}
+          railStyle={railStyle}
+          handleStyle={handleStyle}
+          handle={handleIcon}
+          defaultValue={this.state.value}
+          onAfterChange={this.onAfterChange}
+        />
+        <input
+          type="hidden"
+          name={`vote-for-proposal-${this.props.proposalId}-vote-specification-${this.props.id}`}
+          value={this.state.value}
+          ref={(input) => {
+            this.inputElement = input;
+          }}
+        />
+      </div>
+    );
+  }
+}
+
+type NumberGaugeVoteForProposalProps = {
+  id: string, // the vote specification id
+  instructions: ?string,
+  minimum: ?number,
+  maximum: ?number,
+  nbTicks: ?number,
+  unit: ?string,
+  proposalId: string,
+  voteForProposal: Function,
+  value: number
+};
+
+class NumberGaugeVoteForProposal extends React.Component<*, NumberGaugeVoteForProposalProps, NumberGaugeVoteForProposalState> {
+  props: NumberGaugeVoteForProposalProps;
+
+  state: NumberGaugeVoteForProposalState;
+
+  onAfterChange: Function;
+
+  marks: Object;
+
+  maximum: ?number;
+
+  minimum: ?number;
+
+  inputElement: ?Object;
+
+  step: ?number;
+
+  constructor(props: NumberGaugeVoteForProposalProps) {
+    super(props);
+    this.state = { value: this.props.value };
+    this.onAfterChange = this.onAfterChange.bind(this);
+
+    this.marks = {};
+    this.inputElement = null;
+    this.step = null;
+
+    const minimum = props.minimum;
+    const maximum = props.maximum;
+    const nbTicks = props.nbTicks;
+    const unit = props.unit;
+
+    if (minimum !== undefined && minimum !== null && maximum !== undefined && maximum !== null) {
+      this.marks[`${minimum}`] = {
+        style: markStyle,
+        label: (
+          <div>
+            {minimum} {unit}
```

Comment:
It should be a Translate. In English unit is before.

Reply:
Wait, I think the format is correct. Let me go through examples: 5 min, 10 km, 200 kPa...yeah it's right :D

Reply:
Unit here is meant to be used mainly for currency as far as I understand? Don't we right `$1,000` in English instead of `1,000 $` ? Maybe both are correct, in this case I guess it's ok.